### ISSUE TITLE
Standalone VR Browser for Linux OS

### ISSUE THREAD

**Issue by fire** [Igalia/wolvic#10]
**Is your feature request related to a problem? Please describe.**
Is standard VR browser for PopOS linux out of scope?

**Describe the solution you'd like**
Would like to test wolvic on my Popos Linux for debugging purposes at least.

**Describe alternatives you've considered**
There is no openxr alternative for Linux.

**Additional context**
None.

**Comment by KooIaIa** [comment]
It does say the goal is to be "exclusively for standalone AR and VR headsets" but it doesn't say Android. There are rumors of major companies working on standalone Linux XR headsets. Also a kickstarter for one I believe.

If people get VR working on the Steam Deck next month that does seem like a good starting point. But isn't the bigger issue that OpenXR itself is a mess on Linux and no Linux distros officially support OpenXR out of the box? So in order to target Linux today its a mess of many companies constantly changing non-standard OpenXR solutions? If a new standalone Linux system implements OpenXR and it 'just works' it seems like that's the biggest hurdle.

**Comment by technobaboo** [comment]
There actually is a web engine that fully supports OpenXR and WebXR for Linux, it's a patch to WebKit that works but isn't implemented into any browsers yet: 

https://blogs.igalia.com/ifernandez/2021/01/webxr-webkit/

wait a moment this is from the same people lol, i guess it's expected that it'll work on linux!!

**Comment by technobaboo** [comment]
@KooIaIa openxr isn't that big a mess on Linux, it's certainly better than on Android. Monado isn't the easiest to get working, but when it's installed it's reliable and openxr apps connecting to it work basically 100% of the time. And yea, no distros support OpenXR out of the box because it's just not common right now, but when browsers and apps use it it will start to change.

Assuming you preinstall Monado and it supports the standalone hardware (which I know for a fact [Simula](https://simulavr.com/) is doing) then running a WebXR browser is indeed as plug and play as ideal.

**Comment by guysoft** [comment]
Hey,
I also have an original Vive working on linux, that works pretty well.
I can also run stuff using Proton emulation layer. 
Sadly no browser that I know of has linux builds. I could try and make a linux build it I have a direction how to build this. I can run ``gradlew`` but I assume that would make an APK in the end.
WebXR IMHO would be the way to go.

**Comment by ckorn** [comment]
Can I even get wolvic to work on Valve Index with Windows?

**Comment by rpavlik** [comment]
Debian Bullseye contains a release of Monado that is a conformant OpenXR 1.0 runtime (when used in simulated mode) at release (and so does recent Ubuntu releases, and Monado has CI built packages), it's definitely not a mess on Linux, and at least for simulator mode that's about as "out of the box" as it gets.  SteamVR has some bugs on Linux, but they aren't the only game in town. (They actually never submitted OpenXR conformance on Linux unlike Monado)

From briefly reviewing the code, it's hard for me to see how much of the WebXR and OpenXR-related plumbing stuff is closely coupled to the Java stuff vs the cross platform Firefox/Mozilla stuff.

**Comment by svillar** [comment]
> Monado has come a long way supporting a lot of hardware now and being pretty stable

Right, but the OpenXR runtime is not the problem here. We have actually ported Wolvic to some devices that were using Monado as the OpenXR runtime. The problem is all the other dependencies with Android. Right now Wolvic is basically an AOSP application and getting rid of that dependency would mean almost starting from scratch.
